controllers/playlists.ts

```diff
@@ -65,11 +66,21 @@ export const addTrackToPlaylist = asyncHandler(async (req, res, next) => {
     return;
   }
 
-  const lastPlaylistTrack = await prisma.playlistTrack.findFirst({
+  const currentPlaylistTracks = await prisma.playlistTrack.findMany({
     where: { playlistId: playlistId },
     orderBy: { order: "desc" },
   });
 
+  if (currentPlaylistTracks.length >= MAX_PLAYLIST_LENGTH) {
+    res.status(400).json({
+      success: false,
+      error: `Playlist ${playlistId} is at max length of ${MAX_PLAYLIST_LENGTH}`,
+    });
+    return;
+  }
+  const lastPlaylistTrack =
+    currentPlaylistTracks[currentPlaylistTracks.length - 1];
```

Comment:
We're ordering by `desc` up top so I think this should be the first item?
```suggestion
  const lastPlaylistTrack =
    currentPlaylistTracks[0];
```